lib/galaxy/model/__init__.py

```diff
@@ -1157,7 +1177,7 @@ def is_hda(d):
     return isinstance( d, HistoryDatasetAssociation )
 
 
-class History( object, Dictifiable, UsesAnnotations, HasName ):
+class History( HasTags, Dictifiable, UsesAnnotations, HasName ):
```

Comment:
Hrmm.  Yeah, my concern is that that approach isn't scalable for mixins.  I'll tinker with a few things and see what I can come up with.